enhancements/microshift/optionally-deploy-csi-plugin.md

```diff
@@ -0,0 +1,256 @@
+      `microshift/assets/optional/csi-snapshot-controller/release-$ARCH.json`.
+  - During rebasing, the image digests for LVMS and CSI Snapshotter will be parsed from the LVMS bundle (like they 
+    are now) and the digests will be written to their respective `release-$ARCH.json` files.
+
+- **Mainline Code Changes:**
+  - MicroShift source code will no longer manage LVMS or CSI manifest deployment or its configuration. The 
+    microshift service manager which handles these components will be removed entirely. The following files will be 
+    deleted:
+    - `microshift/pkg/components/csi-snapshot-controller.go`
+    - `microshift/pkg/components/storage.go`
+    - `microshift/pkg/components/render_test.go`, which only tests LVMS parameter rendering
+    - `microshift/pkg/assets/storage.go`
+  - Additionally, certain functions related to managing LVMS will be deleted:
+    - `microshift/pkg/components/render_test.go:startCSIPlugin()`
+    - `microshift/pkg/components/render_test.go:startCSISnapshotterController()`
+  - MicroShift also contains logic to determine if LVM is installed and attempt a best-guess at which volume group, 
+    if any, should be set in the LVMD config. This code will need to be removed and implemented as:
+    - Specifying LVM as an RPM dependency of `microshift-lvms`.
+    - A `microshift-lvms` rpm post-install script that determines what volume group, if any, should be specified.
+
+- **RPM Spec**
+  - New RPM specs will be written for `microshift-lvms` and `microshift-lvms-snapshotting` packages. These have the 
+    following characteristics:
+    - `microshift-lvms-snapshotting` will depend on `microshift-lvms`.
+    - `microshift-lvms` will depend on `lvm2` to ensure host compatibility
+    - `microshift-lvms` will encapsulate LVMS and CSI driver container images and manifests. A post-install script 
+      will attempt a best-guess at a default volume-group by reimplementing the go log in microshift as bash or 
+      python. Not being able to identify a suitable volume-group is not a fatal error, and a warning will be logged 
+      during install. The user will be responsible for properly configuring the lvmd file.
+    - `microshift-lvms-snapshotting` is effectively a wrapper around CSI snapshotting images and manifests. It does 
+      not encapsulate LVMS images or configurations.
+  - It is critical that user-defined LVMD configurations be preserved during upgrades, downgrades, and reinstalls. RPM 
+    post-install logic will be not stomp on existing lvmd configurations. Overwriting the lvmd config poses a 
+    significant threat of orphaning LVM volumes if MicroShift were to change the volume-groups used by lvmd.
+
+### Risks and Mitigations
+
+- N/A
+
+### Drawbacks
+
+- N/A
+
+## Test Plan
+
+There are already tests for RPM and Ostree install processes. This proposal only slightly alters the installation 
+process and thus will require only minimal changes to existing tests.
```

Comment:
We will need to ensure that MicroShift can function normally without storage optional components.
It's true that 100% of tests currently assume the presence of CSI driver.

Reply:
No MicroShift components require dynamically provisioned persistent storage, thus they have no dependency on CSI (LVMS or otherwise).